scripts/common/deploy-system.ts

```diff
@@ -36,10 +37,13 @@ export interface DeploySystemArgs {
 
 const createServiceRegistry = (serviceRegistryInstance: ServiceRegistry) => {
     return async (hash: string, address: string): Promise<void> => {
-        const receipt = await serviceRegistryInstance.addNamedService(hash, address, {
-            gasLimit: '100000',
-        })
-        await receipt.wait()
+        const existingAddress = await serviceRegistryInstance.getServiceAddress(hash)
+        if (existingAddress === constants.AddressZero) {
+            const receipt = await serviceRegistryInstance.addNamedService(hash, address, {
+                gasLimit: '100000',
+            })
+            await receipt.wait()
+        }
```

Comment:
this should update the entry if it does not equal to the existing one

Reply:
I disagree, if it doesn't match then there is something fishy here and needs to be solved manually

Reply:
current basic buy migration will simply fail then

Reply:
because there is an existing entry

Reply:
in general, your comment is correct, if you were using it for `deploy-system` script only. but since it's reused in migration scripts, it needs to be modified

Reply:
Ok, so I guess we need some overrides. parameter, flags allowing certain entries to be overriden